src/librustdoc/html/render.rs

```diff
@@ -246,6 +247,48 @@ struct IndexItem {
     path: String,
     desc: String,
     parent: Option<ast::DefId>,
+    search_type: Option<IndexItemFunctionType>,
+}
+
+/// A type used for the search index.
+struct Type {
+    name: Option<String>,
+}
+
+impl fmt::Display for Type {
+    /// Formats type as {name: $name}.
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(write!(f, "{{"));
+        try!(write!(f, "\"name\":\"{}\"", self.name.clone().unwrap_or("UNKNOWN".to_string())));
```

Comment:
They shouldn't be omitted entirely from the search index, since you might still be able to search them by name. However, the `search_type` of  such items should be serialized as `null` (JS) alltogether. On it!